### PR TITLE
Mitigate hot-loop in SSA

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Compare objects on update, while stripping fields which change on no-op
resource update.

Avoiding https://github.com/kubernetes/kubernetes/issues/100024

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
Code changes seem reasonable, how has this been tested? 
/approve

**Comment by Danil-Grigorev** [comment]
@mdbooth @Fedosin PTAL.

**Comment by Danil-Grigorev** [comment]
@JoelSpeed Yes, for now this PR was tested manually. This PR together with openstack [bits](https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/10) was deployed and didn't end up in endless updates as @mdbooth displayed in his manual testing.

File: controllers/cache.go [hunk 1]
```diff
@@ -104,33 +105,54 @@ type eventToChannelHandler struct {
...
+		}
+
+		// We'd like to compare both objects ignoring changes in managedFields and resourceVersion
+		// Nither of those fields should be applied, so it is ok to strip
+		// Mitigating issue: https://github.com/kubernetes/kubernetes/issues/100024
+
+		old.SetManagedFields(nil)
+		new.SetManagedFields(nil)
+		old.SetResourceVersion("")
+		new.SetResourceVersion("")
```
Comment (mdbooth) [inline controllers/cache.go:146]: These operations add a side-effect that wasn't previously there. We should copy old and new before modifying them.

**Review by mdbooth — CHANGES_REQUESTED** [review]
Have I misunderstood the context here? I don't think this patch directly addresses the issue with SSA patch. Instead, it causes the controller not to call Reconcile for changes to only ManagedFields or ResourceVersion, right?

If so, that seems like a change with a potentially large blast radius. Either way, I think the comment could be more explicit about this (again, assuming I've correctly understood the context!).

File: controllers/cache.go (hunk 1 above)
Comment (Danil-Grigorev) [reply controllers/cache.go]: You are right. It is now updated to work with resource copy.

**Review by mdbooth — APPROVED** [review]
/lgtm

**Pull request merged**